camel-k-maven-plugin must not contain any of these characters \/:"<>|?* but found > @ line 16, column 18

Hello,

After running the following example:

```
git clone https://github.com/abouchama/CamelK-customerAPI
kamel run customer-api.xml \
    --open-api customer-api.json \
    --name customers \
    --dependency camel-undertow \
    --dependency camel-rest \
    --property camel.rest.port=8080 
```
This end with the following error in the camelk operator pod:

> {"level":"info","ts":1573500771.1264646,"logger":"camel-k.trait","msg":"Apply trait: camel"}
> {"level":"info","ts":1573500771.1273205,"logger":"camel-k.trait","msg":"Apply trait: gc"}
> {"level":"info","ts":1573500771.1273396,"logger":"camel-k.trait","msg":"Apply trait: rest-dsl"}
> {"level":"info","ts":1573500771.127784,"logger":"camel-k.maven","msg":"executing: mvn -Dopenapi.spec=/tmp/rest-dsl735823832/0/openapi-spec.json -Ddsl.out=/tmp/rest-dsl735823832/0/openapi-dsl.xml --batch-mode -Dmaven.repo.local=/tmp/artifacts/m2 --settings /tmp/rest-dsl735823832/0/settings.xml","timeout":"3m45s"}
> [INFO] Scanning for projects...
> [ERROR] [ERROR] Some problems were encountered while processing the POMs:
> [ERROR] 'build.plugins.plugin.version' for org.apache.camel.k:camel-k-maven-plugin must not contain any of these characters \/:"<>|?* but found > @ line 16, column 18
>  @ 
> [ERROR] The build could not read 1 project -> [Help 1]
> [ERROR]   
> [ERROR]   The project org.apache.camel.k.integration:camel-k-rest-dsl-generator:1.0.0-M3 (/tmp/rest-dsl735823832/0/pom.xml) has 1 error
> [ERROR]     'build.plugins.plugin.version' for org.apache.camel.k:camel-k-maven-plugin must not contain any of these characters \/:"<>|?* but found > @ line 16, column 18

@astefanutti without this change we got the following error in the camelk operator once we deploy the integration with `kamel run customer.xml`:

```
{"level":"info","ts":1573500771.1264646,"logger":"camel-k.trait","msg":"Apply trait: camel"}
{"level":"info","ts":1573500771.1273205,"logger":"camel-k.trait","msg":"Apply trait: gc"}
{"level":"info","ts":1573500771.1273396,"logger":"camel-k.trait","msg":"Apply trait: rest-dsl"}
{"level":"info","ts":1573500771.127784,"logger":"camel-k.maven","msg":"executing: mvn -Dopenapi.spec=/tmp/rest-dsl735823832/0/openapi-spec.json -Ddsl.out=/tmp/rest-dsl735823832/0/openapi-dsl.xml --batch-mode -Dmaven.repo.local=/tmp/artifacts/m2 --settings /tmp/rest-dsl735823832/0/settings.xml","timeout":"3m45s"}
[INFO] Scanning for projects...
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[ERROR] 'build.plugins.plugin.version' for org.apache.camel.k:camel-k-maven-plugin must not contain any of these characters /:"<>|?* but found > @ line 16, column 18
@
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]
[ERROR] The project org.apache.camel.k.integration:camel-k-rest-dsl-generator:1.0.0-M3 (/tmp/rest-dsl735823832/0/pom.xml) has 1 error
[ERROR] 'build.plugins.plugin.version' for org.apache.camel.k:camel-k-maven-plugin must not contain any of these characters /:"<>|?* but found > @ line 16, column 18
```

--- a/pkg/trait/rest-dsl.go
+++ b/pkg/trait/rest-dsl.go
@@ -225,7 +225,7 @@ func (t *restDslTrait) generateMavenProject(e *Environment) (maven.Project, erro
 			{
 				GroupID:    "org.apache.camel.k",
 				ArtifactID: "camel-k-maven-plugin",
-				Version:    e.RuntimeVersion,
+				Version:    e.CamelCatalog.RuntimeVersion,
 				Executions: []maven.Execution{
 					{
 						Phase: "generate-resources",

